### PR TITLE
If you delete a Pipeline, the run reference leads to a meaningful message

### PR DESCRIPTION
**Pull request by pnaik1**
Closes: #1324 

## Description
#1324 
The page displays the correct error message and a link to navigate to pipeline page
newPipeline

## How Has This Been Tested?

1. Create a Pipeline
2. Create a Run for that Pipeline
3. Delete the Pipeline
4. View the Run on the Global Runs list page
5. Click on the Pipeline link

## Test Impact
no test coverage

## Request review criteria:

Self checklist (all need to be checked):
- [x] The developer has manually tested the changes and verified that the changes work
- [x] The commits have meaningful messages (squashes happen on merge by the bot).
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has added tests or explained why testing cannot be added (unit tests & storybook for related changes)

If you have UI changes: 

- [ ] Included any necessary screenshots or gifs if it was a UI change.
- [x] Included tags to the UX team if it was a UI/UX change.

After the PR is posted & before it merges:
- [ ] The developer has tested their solution on a cluster by using the image produced by the PR to `main`

### CODE REVIEW

**Comment by pnaik1** [comment]
cc @kywalker-rh can you take a look?

**Comment by andrewballantyne** [comment]
@pnaik1 can we not avoid the header "Loading..." & the action items in the top right? ... perhaps we can replace the title with something else? I wonder if the breadcrumbs have value to take you back... maybe we can look into using React Router to send us back to where we came from? 

This is a reasonable stop-gap though for showing an error screen like we did before. 

I suspect Kyle will know more of what we can do.

**Comment by uidoyen** [comment]
@pnaik1 I had quick test this what I see, error msg looks different then your screenshot

Screenshot 2023-07-13 at 11 04 41 AM

**Review by uidoyen — APPROVED** [review]
/lgtm

File: frontend/src/pages/ApplicationsPage.tsx [hunk 1]
```diff
@@ -74,7 +76,7 @@ const ApplicationsPage: React.FC<ApplicationsPageProps> = ({
 
   const renderContents = () => {
     if (loadError) {
-      return (
+      return !errorStatePage ? (
```
Comment (manaswinidas) [inline frontend/src/pages/ApplicationsPage.tsx:79]: Not sure if `errorStatePage` should be checked as a part of `loadError`, but apart from everything else looks fine, including the button navigation for "See all pipelines".
Screenshot 2023-07-13 at 3 38 46 PM

Comment (pnaik1) [reply frontend/src/pages/ApplicationsPage.tsx:79]: So this pipelineLoadError is related to the loadError  , so I have put this condition here , we have a another scenario for this https://github.com/opendatahub-io/odh-dashboard/pull/1497/files#diff-60ce67b4d05babd2f2fa1532ce4290a55e1466f2c59b71bc654e1d6ae78498d9L103 , I referred this one

**Comment by kywalker-rh** [comment]
It seems weird to have the page still say "Loading...", my assumption from seeing this page is we are no longer loading, we've discovered it doesn't exist.

So in that case we should simply call the page "Pipeline not found". If we can remember the pipeline name from the link we should make it "Pipeline '`pipeline name`' not found", and if we can't actually find the name we should remove the '`pipeline name`' part (so in that case it would just be the existing text).

Hopefully that is clear.

File: frontend/src/pages/ApplicationsPage.tsx (hunk 1 above)
Comment (manaswinidas) [reply frontend/src/pages/ApplicationsPage.tsx:79]: This link leads to your own changes

File: frontend/src/concepts/pipelines/content/pipelinesDetails/pipeline/PipelineNotFound.tsx
Comment (andrewballantyne) [inline frontend/src/concepts/pipelines/content/pipelinesDetails/pipeline/PipelineNotFound.tsx:1]: Curious -- why not just display this message instead of all of the DrawerContent/ApplicationPage stuff... if the Pipeline is not there, wouldn't the whole page just be essentially a custom "404 Pipeline Not Found" page?

File: frontend/src/pages/ApplicationsPage.tsx (hunk 1 above)
Comment (andrewballantyne) [reply frontend/src/pages/ApplicationsPage.tsx:79]: Not sure I'm a fan of this -- I get what you're after, but it adds complexity to the ApplicationPage for an existing scenario that already exists. 🤔 It either is an error or we need to support a 404 as a special use-case. Having everyone provide their own potential component will just open us up to a wide variety of ways this is done and make it harder to be consistent.

**Review by andrewballantyne — CHANGES_REQUESTED** [review]
Perhaps we just simplify this and just show nothing but the 404?

Maybe we could mount the ApplicationPage, but make the body the 404 component you made. It would allow us to have some structure. Let me know if you want to talk about this.

I suggested two solutions as it really depends on how each looks. Maybe take screenshots of both or something.

**Comment by pnaik1** [comment]
@andrewballantyne So I made the changes as per your request 
[withActionButton.webm](https://github.com/opendatahub-io/odh-dashboard/assets/99238909/9b8d45ee-f9c7-41c7-8cf1-aab9813ce060)
so you can observe that while loading,  we have action button , So I think its not right to have this button ,  I have added additional condition in https://github.com/opendatahub-io/odh-dashboard/pull/1497/files#diff-153248b4472ef25aa4c01cf4f312bdddef50f7fe6052090df18ddd810437af90R87-R88 
this will also solve when we use the wrong template 
withActionButton
, even here we dont want the action button to be displayed

**Comment by kywalker-rh** [comment]
I'm pretty sure "Wrong pipeline" is your name of a test pipeline, but just want to double check we're not telling a user they have a wrong pipeline.

Also, I'm not totally following your thread, but if its significantly easier to simply not show the `Actions` button, instead of disabling it, I'm fine with that too.

**Comment by pnaik1** [comment]
@kywalker-rh yeah the wrong pipeline is the pipeline name 
So  lets make it little simplified 
I have made the changes and the changes look like this
newPipeline 
this is the thing you had suggested to give a generic name

File: frontend/src/concepts/pipelines/content/pipelinesDetails/pipeline/PipelineDetails.tsx [hunk 2]
```diff
@@ -65,9 +82,12 @@ const PipelineDetails: PipelineCoreDetailsPageComponent = ({ breadcrumbPath }) =
...
               empty={false}
               loaded={pipelineLoad && pipelineTemplateLoaded}
-              loadError={pipelineLoadError || templateLoadError}
+              loadError={templateLoadError}
               headerAction={
-                <PipelineDetailsActions onDelete={() => setDeleting(true)} pipeline={pipeline} />
+                pipelineLoad &&
+                pipelineTemplateLoaded && (
+                  <PipelineDetailsActions onDelete={() => setDeleting(true)} pipeline={pipeline} />
+                )
```
Comment (andrewballantyne) [inline frontend/src/concepts/pipelines/content/pipelinesDetails/pipeline/PipelineDetails.tsx:90]: Curious. Why not take the entire error up to the other block? Is there a reason we would want to show some details and not all 🤔 

Also why the change to the `headerAction` here? What state are you trying to fix here?

Comment (pnaik1) [reply frontend/src/concepts/pipelines/content/pipelinesDetails/pipeline/PipelineDetails.tsx:90]: Because the templateLoad Error shows different error ( here the pipeline is present but the format or yaml file is broken)
withActionButton

**Comment by kywalker-rh** [comment]
> 

Looks good! Thanks for confirming the name. :)

**Review by andrewballantyne — APPROVED** [review]

**Comment by pnaik1** [comment]
I have rebased the branch
